### PR TITLE
remove openssl

### PR DESCRIPTION
**Pull request by drahnr**

### CODE REVIEW

**Comment by drahnr** [comment]
Pending some unit tests to verify the results are equiv.

**Comment by TriplEight** [comment]
if you're waiting travis to be rewritten to gitlab, it kinda [depends](https://github.com/paritytech/sccache/blob/36ccc87af1cc90aba0274d40d67d23aebca19829/.travis.yml#L24) on this PR.

**Comment by drahnr** [comment]
This one depends on me writing some verification tests

File: src/dist/http.rs [hunk 1]
```diff
@@ -304,83 +303,100 @@ mod server {
...
-        let cert: openssl::x509::X509 = builder.build();
-        let cert_pem = cert.to_pem().context("failed to create pem from x509")?;
-        let cert_digest = cert
-            .digest(openssl::hash::MessageDigest::sha256())
-            .context("failed to create digest of x509 certificate")?
-            .as_ref()
-            .to_owned();
 
+    use picky::key::{PublicKey, PrivateKey};
+    use picky::pem::{parse_pem, Pem};
```
Comment (gww-parity) [inline src/dist/http.rs:308]:
```
warning: unused imports: `Pem`, `parse_pem`
   --> src/dist/http.rs:308:22
    |
308 |     use picky::pem::{parse_pem, Pem};
    |                      ^^^^^^^^^  ^^^
    |
    = note: `#[warn(unused_imports)]` on by default
```

File: src/dist/http.rs [hunk 2]
```diff
@@ -304,83 +303,100 @@ mod server {
...
+    use picky::{signature::SignatureAlgorithm, hash::HashAlgorithm};
+    use picky::x509::key_id_gen_method::{KeyIdGenError, KeyIdGenMethod};
+    use picky::x509::{certificate::CertType, csr::Csr};
```
Comment (gww-parity) [inline src/dist/http.rs:311]:
```
warning: unused imports: `certificate::CertType`, `csr::Csr`
   --> src/dist/http.rs:311:23
    |
311 |     use picky::x509::{certificate::CertType, csr::Csr};
    |                       ^^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^
```

File: src/dist/http.rs [hunk 3]
```diff
@@ -304,83 +303,100 @@ mod server {
...
-        let cert_digest = cert
-            .digest(openssl::hash::MessageDigest::sha256())
-            .context("failed to create digest of x509 certificate")?
-            .as_ref()
-            .to_owned();
 
+    use picky::key::{PublicKey, PrivateKey};
+    use picky::pem::{parse_pem, Pem};
+    use picky::{signature::SignatureAlgorithm, hash::HashAlgorithm};
+    use picky::x509::key_id_gen_method::{KeyIdGenError, KeyIdGenMethod};
```
Comment (gww-parity) [inline src/dist/http.rs:310]:
```
warning: unused import: `KeyIdGenError`
   --> src/dist/http.rs:310:42
    |
310 |     use picky::x509::key_id_gen_method::{KeyIdGenError, KeyIdGenMethod};
    |                                          ^^^^^^^^^^^^^

```

File: src/dist/http.rs [hunk 4]
```diff
@@ -304,83 +303,100 @@ mod server {
...
+    use picky::x509::{certificate::CertType, csr::Csr};
+    use picky::x509::name::{GeneralNames, DirectoryName};
+    use picky::x509::date::UTCDate;
+    use picky::x509::certificate::CertificateBuilder;
+    use picky::x509::extension::ExtendedKeyUsage;
+    use picky::x509::extension::KeyUsage;
+    use picky::x509::Extension;
```
Comment (gww-parity) [inline src/dist/http.rs:317]:
```
warning: unused import: `picky::x509::Extension`
   --> src/dist/http.rs:317:9
    |
317 |     use picky::x509::Extension;
    |         ^^^^^^^^^^^^^^^^^^^^^^
```

File: src/dist/http.rs [hunk 5]
```diff
@@ -304,83 +303,100 @@ mod server {
...
+    use picky::x509::Extensions;
+    use sha2::{Sha256, Sha512, Digest};
```
Comment (gww-parity) [inline src/dist/http.rs:319]:
```
warning: unused imports: `Sha256`, `Sha512`
   --> src/dist/http.rs:319:16
    |
319 |     use sha2::{Sha256, Sha512, Digest};
    |                ^^^^^^  ^^^^^^

```

File: src/dist/http.rs [hunk 6]
```diff
@@ -304,83 +303,100 @@ mod server {
...
+    use rsa_pem::KeyExt;
+    use chrono::Datelike;
+    use chrono::Timelike;
+    use std::ops::DerefMut;
+    use std::convert::TryFrom;
```
Comment (gww-parity) [inline src/dist/http.rs:324]:
```
warning: unused import: `std::convert::TryFrom`
   --> src/dist/http.rs:324:9
    |
324 |     use std::convert::TryFrom;
    |         ^^^^^^^^^^^^^^^^^^^^^

```

File: src/dist/http.rs [hunk 7]
```diff
@@ -304,83 +303,100 @@ mod server {
...
+    use std::net::{IpAddr, Ipv4Addr, SocketAddr};
```
Comment (gww-parity) [inline src/dist/http.rs:325]:
```
warning: unused import: `Ipv4Addr`
   --> src/dist/http.rs:325:28
    |
325 |     use std::net::{IpAddr, Ipv4Addr, SocketAddr};
    |                            ^^^^^^^^
```

**Review by gww-parity — CHANGES_REQUESTED** [review]
Regression in compiler warnings introduced with:
https://github.com/paritytech/sccache/pull/7/commits/3f030f3b8ccda40b3af248b20cfcc4bb09ebadd7

**Review by gww-parity — APPROVED** [review]
Now I see no warnings and nice rustfmt on files. Thank you!

In future it will be highly appreciated if I could see rustfmt/style in separate commit, so commits with actual logical changes are easier to focus on. 

Thank you!

**Pull request merged**